lib/dns.js

```diff
@@ -61,17 +61,16 @@ function makeAsync(callback) {
       // The API already returned, we can invoke the callback immediately.
       callback.apply(null, arguments);
     } else {
-      process.nextTick(callMakeAsyncCbNT, callback, arguments);
+      var args = new Array(arguments.length + 1);
+      args[0] = callback;
+      for (var i = 1, a = 0; a < arguments.length; ++i, ++a)
+        args[i] = arguments[a];
+      process.nextTick.apply(null, args);
     }
   };
 }
```

Comment:
DNS callbacks have either two or three arguments so maybe it's cleaner to write this as:

```javascript
return function asyncCallback(arg0, arg1, arg2) {
  if (asyncCallback.immediately) {
    if (arguments.length === 3)
      callback(arg0, arg1, arg2);
    else
      callback(arg0, arg1);
  } else {
    if (arguments.length === 3)
      process.nextTick(callback, arg0, arg1, arg2);
    else
      process.nextTick(callback, arg0, arg1);
  }
};
```

It's still ugly but it avoids allocating an array and going through .apply().